src/plugins/unified_search/public/query_string_input/custom_date_picker_panel/get_documents_time_range.ts

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { lastValueFrom } from 'rxjs';
+import type { DataView } from '@kbn/data-views-plugin/common';
+import type { DataPublicPluginStart } from '@kbn/data-plugin/public';
+
+/**
+ * Fetches the historical time range of the data view documents
+ * @param data
+ * @param dataView
+ * @param abortSignal
+ * @param dslQuery
+ * @public
+ */
+export const getDocumentsTimeRange = async ({
+  data,
+  dataView,
+  abortSignal,
+  dslQuery,
+}: {
+  data: DataPublicPluginStart;
+  dataView: DataView;
+  abortSignal?: AbortSignal;
+  dslQuery?: object;
+}): Promise<{ from: string; to: string } | null> => {
+  if (!dataView?.timeFieldName) {
+    return null;
+  }
+
+  try {
+    const [firstDocumentAt, lastDocumentAt] = await Promise.all([
+      fetchDocumentTimestamp(data, dataView, 'asc', dslQuery, abortSignal),
+      fetchDocumentTimestamp(data, dataView, 'desc', dslQuery, abortSignal),
```

Comment:
I'm interested in feedback about the performance impact. e.g. @nik9000, can you help or guide us to somebody who can? 
So in case, there are no documents found in a time range we provide a link to the user containing the first and last document we find in Elasticsearch. is this fine for large/huge clusters?

Reply:
Going to change it to a single min/max aggregation request.